obs2aptly/src/lib.rs

```diff
@@ -308,18 +309,32 @@ impl ObsContent {
                 changes_version: info.version,
                 aptly_hash: f.aptly_hash(),
             };
-            if info.architecture == "all" {
-                self.binary_indep
-                    .entry(package_name.clone())
-                    .or_insert_with(|| ObsPackage::new(package_name))
-                    .push(deb);
+            let packages = if info.architecture == "all" {
+                &mut self.binary_indep
             } else {
                 self.binary_arch
-                    .entry(info.architecture.to_string())
+                    .entry(info.architecture.to_owned())
                     .or_default()
-                    .entry(package_name.clone())
-                    .or_insert_with(|| ObsPackage::new(package_name))
-                    .push(deb);
+            };
+
+            let mut should_insert_new_package = false;
+
+            if let Some(existing_package) = packages.get_mut(&package_name) {
+                let incoming_version = &deb.source_version;
+                let existing_version = &existing_package.debs[0].source_version;
+                match incoming_version.cmp(existing_version) {
+                    Ordering::Equal => existing_package.push(deb.clone()),
+                    Ordering::Greater => should_insert_new_package = true,
+                    Ordering::Less => (),
+                }
+            } else {
+                should_insert_new_package = true;
+            }
+
+            if should_insert_new_package {
+                let mut package = ObsPackage::new(package_name.clone());
+                package.push(deb);
+                packages.insert(package_name, package);
```

Comment:
hrm this function was really meant to just collect the content in OBS and not do any filtering on it; The syncing policy might want to choose we keep e.g. older arch all packages around in aptly as long as they're still reflected in OBS. So filtering out older verions here seems wrong